pkg/cache/cluster.go

```diff
@@ -600,10 +604,26 @@ func (c *clusterCache) listResources(ctx context.Context, resClient dynamic.Reso
 			listRetry = retry.DefaultRetry
 		}
 
+		if opts.ResourceVersion == "" {
+			opts.ResourceVersion = "0"
+		}
```

Comment:
Why is this necessary?

Reply:
Required by the list options construction function's code, otherwise it would return false as success.

Reply:
https://github.com/argoproj/gitops-engine/pull/617#discussion_r1723570601

Referenced from https://ahmet.im/blog/kubernetes-list-performance/?utm_id=FAUN_Kaptain485_Link_title#list-api-performance-characteristics